src/compas_fab/backends/kinematics/analytical_plan_cartesian_motion.py

```diff
@@ -1,12 +1,25 @@
 import math
 
+import compas
 from compas.geometry import argmin
 
 from compas_fab.backends.interfaces import PlanCartesianMotion
 from compas_fab.backends.kinematics.exceptions import CartesianMotionError
 from compas_fab.backends.kinematics.utils import smallest_joint_angles
 from compas_fab.robots import JointTrajectory
 from compas_fab.robots import JointTrajectoryPoint
+from compas_fab.robots import FrameWaypoints
+from compas_fab.robots import PointAxisWaypoints
+
+from compas_fab.utilities import from_tcf_to_t0cf
+
+if not compas.IPY:
+    from typing import TYPE_CHECKING
+
+    if TYPE_CHECKING:
+        from typing import Optional  # noqa: F401
+        from compas_fab.robots import Robot  # noqa: F401
+        from compas_robots import Configuration  # noqa: F401
```

Comment:
Please remove this from this PR. There's a separate PR for type hinting, and I would really like to address that concretely on the other one because I don't like this huge prefixing that all files will need to deal with ipy's lack of typing.